src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,363 @@
+                                   app::Clusters::Scenes::Structs::ExtensionFieldSet::Type & extensionFieldSet) = 0;
+
+    /// @brief From stored scene (e.g RecallScene), applies EFS values to cluster at transition time
+    /// @param endpoint Endpoint ID
+    /// @param cluster Cluster ID
+    /// @param serialisedBytes ExtensionFieldSet stored in NVM
+    /// @param timeMs Transition time in ms to apply the scene
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR value otherwise
+    virtual CHIP_ERROR ApplyScene(EndpointId endpoint, ClusterId cluster, ByteSpan & serialisedBytes, TransitionTimeMs timeMs) = 0;
+};
+
+class SceneTable
+{
+public:
+    /// @brief struct used to identify a scene in storage by 3 ids, endpoint, group and scene
+    struct SceneStorageId
+    {
+        // Identifies endpoint to which this scene applies
+        EndpointId mEndpointId = kInvalidEndpointId;
+        // Identifies group within the scope of the given fabric
+        SceneGroupID mGroupId = kGlobalGroupSceneId;
+        SceneId mSceneId      = kUndefinedSceneId;
+
+        SceneStorageId() = default;
+        SceneStorageId(EndpointId endpoint, SceneId id, SceneGroupID groupId = kGlobalGroupSceneId) :
+            mEndpointId(endpoint), mGroupId(groupId), mSceneId(id)
+        {}
+        SceneStorageId(const SceneStorageId & storageId) :
+            mEndpointId(storageId.mEndpointId), mGroupId(storageId.mGroupId), mSceneId(storageId.mSceneId)
+        {}
+        CHIP_ERROR Serialize(TLV::TLVWriter & writer) const
+        {
+            TLV::TLVType container;
+            ReturnErrorOnFailure(
+                writer.StartContainer(TLV::ContextTag(kTagSceneStorageIDContainer), TLV::kTLVType_Structure, container));
+
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneEndpointID), static_cast<uint16_t>(this->mEndpointId)));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneGroupID), static_cast<uint16_t>(this->mGroupId)));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneID), static_cast<uint8_t>(this->mSceneId)));
+
+            return writer.EndContainer(container);
+        }
+        CHIP_ERROR Deserialize(TLV::TLVReader & reader)
+        {
+            TLV::TLVType container;
+            ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::ContextTag(kTagSceneStorageIDContainer)));
+            ReturnErrorOnFailure(reader.EnterContainer(container));
+
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneEndpointID)));
+            ReturnErrorOnFailure(reader.Get(this->mEndpointId));
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneGroupID)));
+            ReturnErrorOnFailure(reader.Get(this->mGroupId));
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneID)));
+            ReturnErrorOnFailure(reader.Get(this->mSceneId));
+
+            return reader.ExitContainer(container);
+        }

```

Comment:
This should be in the implementation, not the interface. Is there a reason this needs to be here? It could be functions in SceneTableImpl.cpp internals that takes a SceneStorageId and serialize /deserialize it.